docs/proposals/28-autonomous-shoot-clusters.md

```diff
@@ -0,0 +1,495 @@
+Any day-2 operations like updates or upgrades will be performed by the `gardenlet` and via the `Shoot` API.
+As a consequence, there will be no command in `gardenadm` similar to [`kubeadm upgrade`](https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-upgrade/).
+It is explicitly out of scope to support cluster management without a `gardenlet` connected to a Gardener system (see [this section](#gardenadm-connect) for more details).
+
+### Prerequisites
+
+Similar to [`kubeadm`'s prerequisites](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#before-you-begin), `gardenadm` will assume that the following requirements are met for the machines:
+
+- One or more compatible Linux hosts.
+- Sufficient CPU and RAM (to be specified more concretely later).
+- Full network connectivity between all machines in the cluster (public or private network is fine).
+- [Certain ports](https://kubernetes.io/docs/reference/networking/ports-and-protocols/) must be open.
+- [systemd](https://www.freedesktop.org/wiki/Software/systemd/), [containerd](https://containerd.io/), and its [ctr (client CLI)](https://github.com/projectatomic/containerd/blob/master/docs/cli.md/) must be pre-installed.
+
+### Commands
+
+`gardenadm` will have the following commands:
+
+- [`gardenadm discover`](#gardenadm-discover) to discover the configuration of the Gardener system the autonomous shoot cluster shall be connected to later.
+- [`gardenadm init`](#gardenadm-init) to bootstrap a first control plane node.
+- [`gardenadm token`](#gardenadm-token) to manage bootstrap and discovery tokens for `gardenadm join`.
+- [`gardenadm join`](#gardenadm-join) to bootstrap further control plane nodes, or to bootstrap worker nodes and join them to the cluster.
+- [`gardenadm connect`](#gardenadm-connect) to deploy a `gardenlet` for further cluster management.
+- [`gardenadm version`](#gardenadm-version) to print the `gardenadm` version.
+
+The general flow for bootstrapping an autonomous shoot cluster will be as follows:
+
+1. The user installs `gardenadm` on the first control plane node and runs `gardenadm discover`.
+   This will download relevant configuration from the Gardener system the autonomous shoot cluster shall be later connected to (e.g., `Controller{Registration,Deployment}`s, `CloudProfile`s, etc.).
+   If there is no existing Gardener system yet, the needed configuration must be provided manually.
+2. The downloaded configuration and a `Shoot` manifest are provided to the `gardenadm init` command.
+   This will bootstrap the first control plane node.
+3. Optionally, additional control plane nodes can be joined.
+   In order to do so, the user runs `gardenadm token generate` which generates a random bootstrap token, and then `gardenadm token create <token>` which creates it on the server.
```

Comment:
Can you elaborate on this, what is the difference between generating and creating token? This commands are executed on which node/server (the initial one or any node that already joint the cluster)?

Reply:
This is inspired heavily by `kubeadm token` (see https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-token/), which offers two distinct commands for (1) token generation and (2) token creation in the target (autonomous) shoot cluster.
The assumption is that these commands are executed on the already existing control plane node.